mysql-test/main/partition_exchange.test

```diff
@@ -536,3 +536,11 @@ ALTER TABLE t2 REMOVE PARTITIONING;
 ALTER TABLE t1 EXCHANGE PARTITION pm WITH TABLE t2;
 DROP TABLE t1, t2;
 
+--echo # Test exchange partition with different ROW_FORMAT
+CREATE TABLE t1 (a INT) ROW_FORMAT=DYNAMIC PARTITION BY KEY(a) PARTITIONS 2;
+CREATE TABLE t2 (a INT) ;
+--error ER_PARTITION_EXCHANGE_DIFFERENT_OPTION
+ALTER TABLE t1 EXCHANGE PARTITION p1 WITH TABLE t2;
+ 
+# Cleanup
+DROP TABLE t1, t2;
```

Comment:
Add the following test case, which is a simplified version of one provided by Roel:

```
CREATE TABLE t1 (a INT, PRIMARY KEY(a)) ENGINE=InnoDB PARTITION BY KEY(a) PARTITIONS 2;
CREATE TABLE t2 (a INT, PRIMARY KEY(a)) CHECKSUM=1, ENGINE=InnoDB;
ALTER TABLE t1 EXCHANGE PARTITION p0 WITH TABLE t2;
 
# Cleanup
DROP TABLE t1, t2;
```